src/main/java/de/tum/in/www1/artemis/service/scheduled/ModelingExerciseScheduleService.java

```diff
@@ -66,33 +70,50 @@ public void scheduleRunningExercisesOnStartup() {
             List<ModelingExercise> exercisesToBeScheduled = modelingExerciseRepository.findAllToBeScheduled(ZonedDateTime.now());
             exercisesToBeScheduled.forEach(this::scheduleExercise);
 
-            log.info("Scheduled {} programming exercises.", exercisesToBeScheduled.size());
+            List<ModelingExercise> modelingExercisesWithExam = modelingExerciseRepository.findAllWithEagerExamByExamEndDateAfterDate(ZonedDateTime.now());
+            modelingExercisesWithExam.forEach(this::scheduleExamExercise);
+
+            log.info("Scheduled {} modeling exercises.", exercisesToBeScheduled.size());
+            log.info("Scheduled {} exam programming exercises.", modelingExercisesWithExam.size());
         }
         catch (Exception e) {
-            log.error("Failed to start ProgrammingExerciseScheduleService", e);
+            log.error("Failed to start ModelingExerciseScheduleService", e);
```

Comment:
Since it schedules both exercise types now (?), would it error log 'Failed to start ModelingExerciseScheduleService' even if the exception occurred within the `ProgrammingExerciseScheduleService`?

Reply:
It only schedules modeling exercises